File: app/src/main/java/com/chess/cryptobot/model/response/livecoin/LivecoinHistoryResponse.java
package com.chess.cryptobot.model.response.livecoin;

import com.chess.cryptobot.model.response.HistoryResponse;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZoneOffset;

public class LivecoinHistoryResponse extends LivecoinResponse implements HistoryResponse {
    @SerializedName("type")
    @Expose
    private String type;

    @SerializedName("date")
    @Expose
    private Long date;

    @SerializedName("amount")
    @Expose
    private Double amount;

    @SerializedName("variableAmount")
    @Expose
    private Double variableAmount;

    @SerializedName("fixedCurrency")
    @Expose
    private String fixedCurrency;

    @SerializedName("taxCurrency")
    @Expose
    private String taxCurrency;

    @Override
    public LocalDateTime getHistoryTime() {
        return LocalDateTime.ofEpochSecond(Math.round(date / 1000), 0,
                ZoneOffset.systemDefault().getRules().getOffset(Instant.now()));
    }

    @Override
    public String getHistoryName() {
        return fixedCurrency.equals(taxCurrency) ? fixedCurrency :
                taxCurrency.concat("/").concat(fixedCurrency);
    }

    @Override
    public String getHistoryMarket() {
        return "livecoin";
    }

    @Override
    public Double getHistoryAmount() {
        return amount;
    }

    @Override
    public Double getHistoryPrice() {
        return variableAmount == null ? null : variableAmount / amount;
    }

    @Override
    public String getHistoryAction() {
        return type;
    }

    @Override
    public Integer getProgress() {
        return 0;
    }
}

File: app/src/main/java/com/chess/cryptobot/model/response/livecoin/LivecoinHistoryResponse.kt
package com.chess.cryptobot.model.response.livecoin

import com.chess.cryptobot.model.response.HistoryResponse
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.time.*

class LivecoinHistoryResponse : LivecoinResponse(), HistoryResponse {
    @SerializedName("type")
    @Expose
    override val historyAction: String? = null

    @SerializedName("date")
    @Expose
    private val date: Long? = null

    @SerializedName("amount")
    @Expose
    override val historyAmount: Double? = null

    @SerializedName("variableAmount")
    @Expose
    private val variableAmount: Double? = null

    @SerializedName("fixedCurrency")
    @Expose
    private val fixedCurrency: String? = null

    @SerializedName("taxCurrency")
    @Expose
    private val taxCurrency: String? = null
    override val historyTime: ZonedDateTime?
        get() = ZonedDateTime.ofLocal(LocalDateTime.ofEpochSecond((date!! / 1000.toFloat()).toLong(), 0,
                ZoneOffset.systemDefault().rules.getOffset(Instant.now())), ZoneId.of("Z"), ZoneOffset.UTC)

    override val historyName: String?
        get() {
            return if ((fixedCurrency == taxCurrency)) fixedCurrency else "$taxCurrency/$fixedCurrency"
        }

    override val historyMarket: String
        get() {
            return "livecoin"
        }

    override val historyPrice: Double?
        get() {
            return if (variableAmount == null) null else variableAmount / (historyAmount)!!
        }

    override val progress: Int
        get() {
            return 0
        }
}